contracts/verifiers/Verifier.sol

```diff
@@ -323,20 +317,33 @@ abstract contract Verifier is IVerifier, ContextUpgradeable {
         $._state.processCrossChainProofs(crossChainProofs);
 
         uint256 userIDFromAuthResponse;
-        AuthTypeData storage authTypeData = $._authMethods[authResponse.authType];
+        AuthMethodData storage authMethodData = $._authMethods[authResponse.authMethod];
 
         // 2. Authenticate user and get userID
-        bytes32 expectedNonce = keccak256(abi.encode(sender, responses)) &
-            0x0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
-
-        userIDFromAuthResponse = authTypeData.validator.verify(
-            authResponse.proof,
-            authTypeData.params,
+        IAuthValidator.AuthResponseField[] memory authResponseFields;
+        (userIDFromAuthResponse, authResponseFields) = authMethodData.validator.verify(
             sender,
-            $._state,
-            expectedNonce
+            authResponse.proof,
+            authMethodData.params
         );
 
+        if (
+            keccak256(abi.encodePacked(authResponse.authMethod)) ==
+            keccak256(abi.encodePacked("authV2"))
+        ) {
+            if (
+                authResponseFields.length > 0 &&
+                keccak256(abi.encodePacked(authResponseFields[0].name)) ==
+                keccak256(abi.encodePacked("challenge"))
+            ) {
+                bytes32 expectedNonce = keccak256(abi.encode(sender, responses)) &
+                    0x0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
```

Comment:
I think it should be 0x00FFFF... if we want to zero out the first byte.

Reply:
Only with the first position as 0 that is 4 bits we have 252 bits remaining for the calculation that fits in a field number. But I could do it with 1 byte if you think it's better an we we'll use 248 bits then.